lib/improver/utilities/spatial.py

```diff
@@ -114,58 +129,59 @@ def convert_distance_into_number_of_grid_cells(
                 distance in metres.
 
     Raises:
-        ValueError:
-            If the projection is not "equal area" (proxied by projection_x/y
-            spatial coordinate names).
+        ValueError: If a negative distance is provided
+        ValueError: If the projection is not equal-area
         ValueError:
             If the distance in grid cells is larger than the maximum dimension
             of the rectangular domain (measured across the diagonal).  Needed
             for neighbourhood processing.
-        ValueError:
-            If the distance in grid cells is zero.
-        ValueError:
-            If the distance in grid cells is negative.  (Assuming the distance
-            argument is positive, this indicates one or more spatial axes are
-            not correctly ordered.)
+        ValueError: If the distance in grid cells is zero.
         Value Error:
             If max_distance_in_grid_cells is set and the distance in grid cells
             exceeds this value.  Needed for neighbourhood processing.
     """
-    try:
-        x_coord = cube.coord("projection_x_coordinate").copy()
-        y_coord = cube.coord("projection_y_coordinate").copy()
-    except CoordinateNotFoundError:
-        raise ValueError("Invalid grid: projection_x/y coords required")
-    x_coord.convert_units("metres")
-    y_coord.convert_units("metres")
-    max_distance_of_domain = np.sqrt(
-        (x_coord.points.max() - x_coord.points.min())**2 +
-        (y_coord.points.max() - y_coord.points.min())**2)
+    d_error = "Distance of {}m".format(distance)
+    zero_distance_error = ("{} gives zero cell extent".format(d_error))
+    if distance == 0:
+        raise ValueError(zero_distance_error)
+    if distance < 0:
+        raise ValueError("Please specify a positive distance in metres")
+
+    # calculate grid spacing or raise error if cube is not equal area
+    check_if_grid_is_equal_area(cube)
+    grid_spacing_metres = calculate_grid_spacing(cube)
+
+    # check required distance isn't greater than the size of the domain
+    # (note: this implicitly assumes equal x- and y-spacing)
+    def calculate_domain_extent(coord):
+        """Calculates the coordinate extent in metres"""
+        new_coord = coord.copy()
+        new_coord.convert_units('metres')
+        return max(new_coord.points) - min(new_coord.points)
+
+    x_extent_metres = calculate_domain_extent(cube.coord(axis='x'))
+    y_extent_metres = calculate_domain_extent(cube.coord(axis='y'))
+    max_distance_of_domain = np.sqrt(x_extent_metres**2 + y_extent_metres**2)
```

Comment:
I'm glad all of this will be moved in IMPRO-1402. This function should do what its name suggests and not other unexpected things.